docs-src/concepts/what-is-a-codec-server.md

```diff
@@ -2,42 +2,55 @@
+You can create a custom [Payload Codec](/prod-readiness-context/data-encryption) with your encoding logic (such as encryption and/or compression), and apply it to the data processed in your Workflows.
 
-![](/img/tctl-diagram-codec-server.svg)
+Using a custom Payload Codec in your Codec Server enables encoding and decoding data remotely through the endpoints that you expose on the Codec Server.
+
+A Codec Server follows the Temporal [Codec Server Protocol](https://github.com/temporalio/samples-go/tree/main/codec-server#codec-server-protocol).
+It implements two endpoints:
+
+- `POST /encode`
+- `POST /decode`
+
+Each endpoint receives and responds with a JSON body that has a `payloads` property with an array of Payloads.
+The endpoints run the Payloads through a [Payload Codec](/concepts/what-is-a-data-converter#payload-codec) before returning them.
+
+Most SDKs provide example Codec Server implementation samples, listed here:
 
-## Use case: tctl
+- [Go](https://github.com/temporalio/samples-go/tree/main/codec-server)
+- [Java](https://github.com/temporalio/sdk-java/tree/master/temporal-remote-data-encoder)
+- [Python](https://github.com/temporalio/samples-python/blob/main/encryption/codec_server.py)
+- [TypeScript](https://github.com/temporalio/samples-typescript/blob/main/encryption/src/codec-server.ts)
 
-Suppose that you want to view Workflow History.
-This information needs to be decoded before it can be viewed.
+#### Usage
 
-You can use [tctl workflow showid](/tctl-v1/workflow#show) to view a Workflow Execution Event History.
+When using tctl or the Web UI to perform some operations on a Workflow Execution, you can configure the exposed Codec Server endpoints to remotely encode data sent to the Temporal Server and decode data received from the Temporal Server.
 
-```bash
-tctl workflow showid <workflowID>
-```
+When you apply custom encoding with encryption or compression on your Workflow data, it is stored in the encrypted/compressed format on the Temporal Server.
 
-With a Codec Server, Payloads that are part of the Event History will be sent to the Codec Server to be decoded before being deserialized by the Default Data Converter and displayed in your terminal.
+For example, the following objects are persisted in your encoded format in the Workflow Execution Event History:
 
-- [How to configure tctl with a Codec Server](/security/how-to-set-up-codec-server#configure-tctl)
+- Inputs and outputs/results in your [Workflow](/concepts/what-is-a-workflow-execution), [Activity](/concepts/what-is-an-activity-execution), and [Child Workflow](/concepts/what-is-a-child-workflow-execution)
+- [Signal](/concepts/what-is-a-signal) inputs
+- Results of [Local Activity](/concepts/what-is-a-local-activity), [Side Effects](/concepts/what-is-a-side-effect)
+- [Query](/concepts/what-is-a-query) results
+- [Memo](/concepts/what-is-a-memo)
+- [Application errors and failures](/kb/failures).
```

Comment:
Not exactly, Failures are not encoded by default, users need to explicitly enable it. I would at least add a note about it.

Reply:
Added a note "  Note that failures are not encoded by default; you must enable encoding on failures.
"